tests/compiler/inlining.ts

```diff
@@ -6,3 +8,55 @@ export function test(): i32 {
 }
 
 assert(test() == 3);
+
+// Function inlining
+
+@inline
+function func_ii(a: i32): i32 {
+  if (a == 42) return 1;
+  return a < 42 ? 2 : 3;
+}
+
+@inline
+function func_ii_opt(a: i32 = 0): i32 {
+  return a;
+}
+
+@inline
+function func_ii_loc(a: i32): i32 {
+  var b = a;
+  var e: i32;
+  if (true) {
+    let c = b;
+    let d = c;
+    e = d + 1;
+  }
+  return e;
+}
+
+@inline
+function func_iv(a: i32): void {
+}
+
+class Foo {
+  @inline
+  method_this(): Foo {
+    return this;
+  }
```

Comment:
Also inline static methods?

Reply:
Static functions should inline just as normal top-level functions, as these are basically the same, just with a fancy name including a dot. Going to add a test to be sure.